File: remote/src/main/java/com/kelsos/mbrc/annotations/Queue.java
package com.kelsos.mbrc.annotations;

import android.support.annotation.StringDef;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

public class Queue {
  public static final String NEXT = "next";
  public static final String LAST = "last";
  public static final String NOW = "now";

  private Queue() {
    //no instance
  }

  @StringDef({
      NEXT,
      LAST,
      NOW
  })
  @Retention(RetentionPolicy.SOURCE)
  public @interface Action {
  }
}

File: mbrc-lib/src/main/kotlin/com/kelsos/mbrc/annotations/Queue.kt
package com.kelsos.mbrc.annotations

import android.support.annotation.StringDef

object Queue {
    const val NEXT = "next"
    const val LAST = "last"
    const val NOW = "now"
    const val UNDEF = "undef"

    @StringDef(NEXT, LAST, NOW, UNDEF)
    @kotlin.annotation.Retention(AnnotationRetention.SOURCE)
    annotation class Action
}
